src/lib.rs

```diff
@@ -67,21 +73,26 @@ impl Drop for IRObject {
 }
 
 impl IRObject {
-    pub fn create_from_dxil(
-        compiler: &IRCompiler,
-        bytecode: &[u8],
-    ) -> Result<IRObject, Box<dyn std::error::Error>> {
+    pub fn create_from_dxil(compiler: &IRCompiler, bytecode: &[u8]) -> IRObject {
         unsafe {
-            let me = compiler.funcs.IRObjectCreateFromDXIL(
+            let me = NonNull::new(compiler.funcs.IRObjectCreateFromDXIL(
                 bytecode.as_ptr(),
                 bytecode.len(),
                 bindings::IRBytecodeOwnership::None,
```

Comment:
Shouldn't this be `Copy` since the returned `IRObject` doesn't actually have a lifetime borrow on `create_from_dxil()`?

We could also make a `Cow` that is stored in `IRObject` or something complex like that.

Reply:
I'll set it to copy